keymaster/ta/keystore_ta.c

```diff
@@ -147,12 +147,16 @@ static keymaster_error_t TA_configure(TEE_Param params[TEE_NUM_PARAMS])
 	uint8_t *in_end = NULL;
 	size_t  in_size = 0;
 	uint8_t *out = NULL;
+	uint8_t *out_end = NULL;
+	size_t out_size = 0;
 	keymaster_error_t res = KM_ERROR_OK;
 
 	in = (uint8_t *)params[0].memref.buffer;
 	in_size = (size_t)params[0].memref.size;
 	in_end = in + in_size;
 	out = (uint8_t *)params[1].memref.buffer;
+	out_size = (size_t)params[1].memref.size; /* limited to 8192 */
+	out_end = out + out_size;
```

Comment:
Shouldn't we be also adding out of bound check for out ?

Reply:
I just re-read the commit...so you are going t do that in next commit. Ignore this comment.